### ISSUE TITLE
[BUG]: dist-types/index.d.ts is not present in the published NPM package

### ISSUE THREAD

**Issue by maxnachlinger** [octokit/plugin-retry.js#405]
### What happened?

Hey awesome maintainers! After upgrading from 4.1.1 to 4.1.2 I noticed the following tsc error:
```shell
 error TS7016: Could not find a declaration file for module '@octokit/plugin-retry'. '/some-path-here/node_modules/.pnpm/@octokit+plugin-retry@4.1.2_@octokit+core@4.2.0/node_modules/@octokit/plugin-retry/dist-node/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/octokit__plugin-retry` if it exists or add a new declaration (.d.ts) file containing `declare module '@octokit/plugin-retry';`

2 import { retry } from '@octokit/plugin-retry';
```
Having a look in the tgz on npm I only see a `dist-node` directory, but the package.json exports the types in `dist-types/index.d.ts`

### Versions

"@octokit/core": "4.2.0"
"@octokit/plugin-retry": "4.1.2",
"@octokit/rest": "19.0.7",
node 16.19.1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by oscard0m** [comment]
I started checking this:

Here are two logs comparing a release with types and without types:
### [Release `4.1.1`](https://github.com/octokit/plugin-retry.js/actions/runs/4019737690/jobs/6906921514)

```
@pika/pack build v0.3.7
[1/6] Validating source...
[2/6] Preparing pipeline...
      ❇️  pkg/
      » tsconfig.json [compilerOptions.target] should be "ES2020", but found "ES2018". You may encounter problems building.
[3/6] Running @pika/plugin-ts-standard-pkg...
      📝  pkg/dist-src/index.js [esnext]
      📝  pkg/dist-types/index.d.ts [types]
      » Linting with standard-pkg...
[4/6] Running @pika/plugin-build-node...
      📝  pkg/dist-node/index.js [main]
[5/6] Running @pika/plugin-build-web...
      📝  pkg/dist-web/index.js [module]
[6/6] Finalizing package...
      » copying LICENSE...
      » copying README.md...
      📝  pkg/package.json
      📦  pkg/
```

```
npm notice 📦  @octokit/plugin-retry@4.1.1
npm notice === Tarball Contents === 
npm notice 1.1kB LICENSE                      
npm notice 2.5kB README.md                    
npm notice 2.9kB dist-node/index.js           
npm notice 6.2kB dist-node/index.js.map       
npm notice 677B  dist-src/error-request.js    
npm notice 921B  dist-src/index.js            
npm notice 32B   dist-src/version.js          
npm notice 1.4kB dist-src/wrap-request.js     
npm notice 105B  dist-types/error-request.d.ts
npm notice 401B  dist-types/index.d.ts        
npm notice 40B   dist-types/version.d.ts      
npm notice 91B   dist-types/wrap-request.d.ts 
npm notice 2.9kB dist-web/index.js            
npm notice 6.0kB dist-web/index.js.map        
npm notice 1.2kB package.json
```

### [Release `4.1.2`](https://github.com/octokit/plugin-retry.js/actions/runs/4212974120/jobs/7312375151)

```
@pika/pack build v0.3.7
[1/6] Validating source...
[2/6] Preparing pipeline...
      ❇️  pkg/
      » tsconfig.json [compilerOptions.target] should be "ES2020", but found "ES2018". You may encounter problems building.
[3/6] Running @pika/plugin-ts-standard-pkg...
      📝  pkg/dist-src/index.js [esnext]
      📝  pkg/dist-types/index.d.ts [types]
      » Linting with standard-pkg...
[4/6] Running @pika/plugin-build-node...
      📝  pkg/dist-node/index.js [main]
[5/6] Running @pika/plugin-build-web...
      📝  pkg/dist-web/index.js [module]
[6/6] Finalizing package...
      » copying LICENSE...
      » copying README.md...
      📝  pkg/package.json
      📦  pkg/
```

```
npm notice 📦  @octokit/plugin-retry@4.1.2
npm notice === Tarball Contents === 
npm notice 1.1kB LICENSE           
npm notice 2.5kB README.md         
npm notice 2.9kB dist-node/index.js
npm notice 1.2kB package.json   
```

---

Tagging @octokit/js-community @octokit/js just in case somebody knows where the issue could be while I check it.

**Comment by oscard0m** [comment]
Ok, so... the root cause of this issue is the following one, I think:

Apparently, since `npm 9` [the file pattern matching for `npm publish` has changed, but nothing was reported in their Changelog (I'm assuming it's an unintended change)](https://github.com/npm/cli/issues/6164).

The generated `package.json` under `pkg/` folder after running `pika-pack build` contains the following:

```json
"files": [
    "dist-*/",
    "bin/"
  ],
```

The fix would be as easy as adding `**` to the file patterns until `npm` gives a solution to this:

```json
"files": [
    "dist-*/**",
    "bin/**"
  ],
```
Since these lines are auto-generated and `pika-pack` is unmaintained, I guess we need to decide what to do here. 

I guess the quick solution is to use `npm@8` for `semantic-release` until we find a proper solution for everything.

What do you think @octokit/js @octokit/js-community ?

**Comment by kfcampbell** [comment]
@oscard0m thank you for the investigation here! That's fascinating. I know npm 9 was released in October; I wonder why this is just happening now. Perhaps it's related to a shift in the LTS version in our [node setup](https://github.com/octokit/plugin-retry.js/blob/main/.github/workflows/release.yml#L14)? 

Pinning down a date would help with the solution I think: if we started using npm 9 to release immediately before the issue was reported, a rollback seems appropriate. If we've been using npm 9 since October, rolling back a major version may have additional complications to think about.

**Comment by wolfy1339** [comment]
We used to explicitly define a requirement on `semantic-release` in our `devDependencies`, until a new version of `semantic-release` broke compatibility with a version of node we were supporting at the time, and I believe that version introduced `npm@9`

Would we be able to patch the files afterwards pika has built the package before `semantic-release` publishes a release?

**Comment by oscard0m** [comment]
@kfcampbell opened a PR with @wolfy1339 's proposal as a hot fix (with some question marks) and opened a discussion to talk about a strategy for a long-term solution: https://github.com/octokit/octokit.js/discussions/2403

**Comment by Ddupasquier** [comment]
Thanks for showing me to this issue @wolfy1339. I patiently await updates.

**Comment by Ddupasquier** [comment]
I updated to the newest release, but I'm getting a new error:

ReferenceError: global is not defined
    at node_modules/node-fetch/lib/index.mjs (index.mjs:657:16)
    at __init (chunk-J43GMYXM.js?v=5cdb3d67:8:56)
    at node_modules/@octokit/request/dist-web/index.js (index.js:5:14)
    at __init (chunk-J43GMYXM.js?v=5cdb3d67:8:56)
    at node_modules/@octokit/core/dist-web/index.js (index.js:7:14)
    at __init (chunk-J43GMYXM.js?v=5cdb3d67:8:56)
    at index.js:76:15
    
I'm unsure if maybe my syntax is incorrect or if this is an oversight of the new release.
    Here is the function in which I am making my request to the api:

```js
import { Octokit } from 'octokit';
const octokit = new Octokit({
	auth: import.meta.env.VITE_GITHUB_TOKEN
});
export async function load() {
	const { data } = await octokit.request('GET /repos/{owner}/{repo}/issues', {
		owner: 'Ddupasquier',
		repo: 'stories-client'
	});
      if (data) {
            return data;
      } else {
            throw new Error('Failed to load data')
      }
}
```

If this requires that I make a new issue then I'm happy to do it. Just let me know.

**Comment by wolfy1339** [comment]
That is another unrelated issue. I believe it is an issue with how `node-fetch` patches in `fetch()`

There is an issue somewhere that had a workaround (it might not be this particular octokit package)

**Comment by Ddupasquier** [comment]
Dang xD I'm just gonna table this project for a bit then. Any chance you know off the top of your head where I could track a ticket for the issue?

**Comment by wolfy1339** [comment]
I found the issue that is tracking this on our side
https://github.com/octokit/core.js/issues/547

**Comment by Ddupasquier** [comment]
Awesome! Thanks a ton!

**Comment by oscard0m** [comment]
Here's a comparison of exported files from 4.1.2 and 4.1.3:

### [Published files in `4.1.2`](https://www.npmjs.com/package/@octokit/plugin-retry/v/4.1.2?activeTab=explore):
![image](https://user-images.githubusercontent.com/2574275/222953385-2b0f3c0e-34c9-4225-829d-5920e05711ac.png)

### [Published files in `4.1.2`](https://www.npmjs.com/package/@octokit/plugin-retry/v/4.1.3?activeTab=explore):
![image](https://user-images.githubusercontent.com/2574275/222953379-bf7d229b-c257-491f-81ef-05b4c9679e0f.png)

**Comment by Ddupasquier** [comment]
Omg, if this is actually done I will buy everyone involved a beer.

**Comment by oscard0m** [comment]
@maxnachlinger @Ddupasquier did you have a chance to try this? Did your problems get solved?

**Comment by maxnachlinger** [comment]
@oscard0m forgive me for not circling back. Your dazzling fix solved the issue! :)

**Comment by oscard0m** [comment]
> @oscard0m forgive me for not circling back. Your dazzling fix solved the issue! :)

No problem! I was just going through my old notifications and I just wanted to double-check everything was working again.

I'm happy we unblocked you @maxnachlinger :)

**Comment by Ddupasquier** [comment]
@oscard0m In order to give you an accurate answer I will have to revisit the component and remember what I was even doing haha. On first view of the code... No. But this could be a syntax issue on my part or maybe my token expired.
I'll get back to you when I know more.

**Comment by Ddupasquier** [comment]
@oscard0m AH! So if you check the 'unrelated comments' a little ways up between @wolfy1339 and I, it seems that the issue from the octokit side is solved.

https://github.com/octokit/plugin-retry.js/issues/405#issuecomment-1454299651

But success in my project is still blocked haha.
Wolfy tipped me off the where octokit is tracking this issue as well:
https://github.com/octokit/core.js/issues/547

**Comment by oscard0m** [comment]
> 

Yep, the `@octokit-next` is blocking quite a few things right now... I'm sorry to hear that. But I assume this issue can remain closed.

**Comment by Ddupasquier** [comment]
Absolutely. I appreciate everyone's dillogence!
